### PR TITLE
Refactor proxy HTTP handlers to make it more testable

### PR DESCRIPTION
**Pull request by s-radyuk**
EES-3872

### CODE REVIEW

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -45,28 +63,27 @@ func getLoadTestType(r *http.Request) apisLoadTestV1.LoadTestType {
 	return apisLoadTestV1.LoadTestType(r.FormValue(backendType))
 }
 
-//CreateLoadTestHandler creates loadtest CR on POST request
-func CreateLoadTestHandler(loadTestClient loadTestV1.LoadTestInterface, maxLoadTestsRun int) http.Handler {
+//Create creates loadtest CR on POST request
+func (p *Proxy) Create() http.Handler {
```
Comment (aleksandrzhiliaev) [inline pkg/proxy/proxy.go:67]: All proxy methods should expect writer and request as input like this:
```suggestion
func (p *Proxy) Create(w http.ResponseWriter, r *http.Request) {
```

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -110,43 +127,12 @@ func CreateLoadTestHandler(loadTestClient loadTestV1.LoadTestInterface, maxLoadT
...
-	}
-
-	return tf
+	})
 }
 
-//DeleteLoadTestHandler deletes load test CR
-func DeleteLoadTestHandler(loadTestClient loadTestV1.LoadTestInterface) http.Handler {
+//Delete deletes load test CR
+func (p *Proxy) Delete() http.Handler {
```
Comment (aleksandrzhiliaev) [inline pkg/proxy/proxy.go:135]:
```suggestion
func (p *Proxy) Delete(w http.ResponseWriter, r *http.Request)  {
```

File: pkg/proxy/proxy.go [hunk 3]
```diff
@@ -167,10 +153,8 @@ func DeleteLoadTestHandler(loadTestClient loadTestV1.LoadTestInterface) http.Han
 	})
 }
 
-//GetLoadTestHandler returns the loadtest CR info.
-//For now it returns only created namespace,
-//will be extended when more business logic added
-func GetLoadTestHandler(loadTestClient loadTestV1.LoadTestInterface) http.Handler {
+//Get returns the loadtest CR info
+func (p *Proxy) Get() http.Handler {
```
Comment (aleksandrzhiliaev) [inline pkg/proxy/proxy.go:157]:
```suggestion
func (p *Proxy) Get(w http.ResponseWriter, r *http.Request)  {
```

File: pkg/proxy/proxy.go [hunk 4]
```diff
@@ -45,76 +63,168 @@ func getLoadTestType(r *http.Request) apisLoadTestV1.LoadTestType {
...
+
+//Delete deletes load test CR
+func (p *Proxy) Delete(w http.ResponseWriter, r *http.Request) {
+	logger := mPkg.GetLogger(r.Context())
+
+	ctx, cancel := context.WithTimeout(context.Background(), loadtest.KubeTimeout)
+	defer cancel()
+
+	ltID := chi.URLParam(r, loadTestID)
+	logger.Debug("Deleting loadtest", zap.Any("ltID", ltID))
```
Comment (vgarvardt) [inline pkg/proxy/proxy.go:139]:
```suggestion
	logger.Debug("Deleting loadtest", zap.String("ltID", ltID))
```

File: pkg/proxy/proxy.go [hunk 5]
```diff
@@ -45,76 +63,168 @@ func getLoadTestType(r *http.Request) apisLoadTestV1.LoadTestType {
...
+
+//Get returns the loadtest CR info
+func (p *Proxy) Get(w http.ResponseWriter, r *http.Request) {
+	logger := mPkg.GetLogger(r.Context())
+
+	ctx, cancel := context.WithTimeout(context.Background(), loadtest.KubeTimeout)
+	defer cancel()
+
+	ltID := chi.URLParam(r, loadTestID)
+	logger.Debug("Retrieving info for loadtest", zap.Any("ltID", ltID))
```
Comment (vgarvardt) [inline pkg/proxy/proxy.go:159]:
```suggestion
	logger.Debug("Retrieving info for loadtest", zap.String("ltID", ltID))
```

File: pkg/proxy/proxy.go [hunk 6]
```diff
@@ -45,76 +63,168 @@ func getLoadTestType(r *http.Request) apisLoadTestV1.LoadTestType {
...
 	})
 }
 
-//FileToString converts file to string
-func FileToString(f io.ReadCloser) (string, error) {
+//GetLogs returns the loadtest CR info
+func (p *Proxy) GetLogs(w http.ResponseWriter, r *http.Request) {
+	logger := mPkg.GetLogger(r.Context())
+	ltID := chi.URLParam(r, loadTestID)
+	logger.Info("Retrieving info for loadtest", zap.Any("ltID", ltID))
```
Comment (vgarvardt) [inline pkg/proxy/proxy.go:188]:
```suggestion
	logger.Info("Retrieving info for loadtest", zap.String("ltID", ltID))
```

File: pkg/proxy/proxy.go [hunk 7]
```diff
@@ -145,61 +255,18 @@ func ConvertTestName(n string) string {
...
-			logger.Error("Could not get load test info with error:", zap.Error(err))
-
-			if k8sErr, ok := err.(k8sAPIErrors.APIStatus); ok && k8sErr.Status().Code == http.StatusNotFound {
-				render.Render(w, r, cHttp.ErrResponse(http.StatusNotFound, err.Error()))
-				return
-			}
+func doRequest(req *restClient.Request) ([]byte, error) {
+	stream, err := req.Stream(context.Background())
+	if err != nil {
+		return []byte{}, fmt.Errorf("error in opening stream: %w", err)
```
Comment (vgarvardt) [inline pkg/proxy/proxy.go:261]: As I remember `nil` is valid return value for any slice type and in this case it is more efficent as you do not need to create ampty slice

```suggestion
		return nil, fmt.Errorf("error in opening stream: %w", err)
```

File: pkg/proxy/proxy.go [hunk 8]
```diff
@@ -145,61 +255,18 @@ func ConvertTestName(n string) string {
...
+	}
+	defer stream.Close()
 
-			render.Render(w, r, cHttp.ErrResponse(http.StatusInternalServerError, err.Error()))
-			return
-		}
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, stream)
+	if err != nil {
+		return []byte{}, errors.New("error in copy information from podLogs to buf")
```
Comment (vgarvardt) [inline pkg/proxy/proxy.go:268]:
```suggestion
		return []byte{}, fmt.Errorf("error in copy information from podLogs to buf: %w", err)
```

**Comment by vgarvardt** [comment]
👍

**Pull request merged**